src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
 
+export interface Figure {
+  shape : Shapes,
+  color : Colors,
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
 
-}
+  constructor(
+    public color: string,
+    public a : number,
+    public b : number,
+    public c : number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.a + this.c <= this.b
+      || this.b + this.c <= this.a) {
+      throw Error('Incorrect length');
```

Comment:
maybe better use 2 types errors. User should understand where exactly he was wrong